optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingGenerator.java

```diff
@@ -446,31 +454,75 @@ private void createTalkList(ConferenceSolution solution, int talkListSize) {
                     break;
                 }
             }
+            Set<String> requiredRoomTagSet = new LinkedHashSet<>();
+            Set<String> preferredRoomTagSet = new LinkedHashSet<>();
+            Set<String> prohibitedRoomTagSet = new LinkedHashSet<>();
+            Set<String> undesiredRoomTagSet = new LinkedHashSet<>();
+            setRoomTags(requiredRoomTagSet, preferredRoomTagSet, prohibitedRoomTagSet, undesiredRoomTagSet);
+            Set<String> mutuallyExclusiveTagSet = new LinkedHashSet<>();
+            if (random.nextDouble() < 0.025) {
+                mutuallyExclusiveTagSet.add(mutuallyExclusiveTagList.get(random.nextInt(mutuallyExclusiveTagList.size())));
+            }
```

Comment:
Same here (vertical density). Put these 4 lines just before setMutuallyExclusiveTalksTagSet() call.
And the next 4 lines just before the setPrerequisiteTalkSet() call.
It makes it easier to read and just focus on the bit that is going on.